workspaces/util/src/__tests__/squashWhitespace-test.ts

```diff
@@ -0,0 +1,35 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { squashWhitespace } from "../squashWhitespace";
+
+describe("squashWhitespace", () => {
+  it("removes whitespaces between words if it exists", () => {
+    expect(squashWhitespace("hello   world")).toBe("hello world");
+  });
+
+  it("removes whitespaces at the end of the string", () => {
+    expect(squashWhitespace("hello world        ")).toBe("hello world");
+  });
+
+  it("removes whitespaces at the beginginng of the string", () => {
+    expect(squashWhitespace("         hello world")).toBe("hello world");
+  });
+
+  it("nothing should be changed", () => {
+    expect(squashWhitespace("hello world !")).toBe("hello world !");
+  });
+
+  it("empty string", () => {
+    expect(squashWhitespace("")).toBe("");
+  });
+
+  it("respects case", () => {
+    expect(squashWhitespace(" Hello   World  ")).toBe("Hello World");
+  });
+
+  it("works with emojis", () => {
+    expect(squashWhitespace("🍎   apple")).toBe("🍎 apple");
+    expect(squashWhitespace("🍌banana")).toBe("🍌banana");
+    expect(squashWhitespace("grapes 🍇 🍇   ")).toBe("grapes 🍇 🍇");
+  });
+});
```

Comment:
A couple more test cases to consider:

* a string that is entirely whitespace
* a string that has a single whitespace character that's not a space, e.g. "hello\nworld" --> "hello world"
* a string that has whitespace at the beginning, end, and repeated whitespace in the middle (not sure this was explicitly covered so far)